### PR TITLE
Redirect to the profile update form after they log in on user deep link

### PR DESCRIPTION
**Pull request by matheusmendes-cb**
closes 
#837 

fixes 

## What changed? Why?

To redirect the user back to the profile update form if we deep link a user to update their profile and they arent logged in we have to persist this data somewhere, in this case it was simpler to persist this queryParam on the URL, so now we redirect to the login url persisting the queryParam and once the user is authenticated, we get check if the user has the correct queryParam and if so, we redirect him back to the deep link url.

## How has it been tested?

- [x] Locally
- [x] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

### CODE REVIEW

File: src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx [hunk 1]
```diff
@@ -15,11 +17,15 @@ export default function UserActionOptInSWCDeepLink() {
   const urls = useIntlUrls()
   const router = useRouter()
   const session = useSession()
+  const { queryString } = useQueryParamState({
+    queryParamKey: OPEN_UPDATE_USER_PROFILE_FORM_QUERY_PARAM_KEY,
```
Comment (travisbloom-cb) [inline src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx:21]: Thoughts on this being generic so you can pass any arbirary redirect path you want here? Like `redirectAfterLogin=/foo/bar?baz=1`

File: src/types/index.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ export type PageProps<Params = object> = {
   params: Params & {
     locale: SupportedLocale
   }
+  searchParams?: { [key: string]: string | string[] | undefined }
```
Comment (travisbloom-cb) [inline src/types/index.ts:9]: TIL

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx (hunk 1 above)
Comment (travisbloom-cb) [reply src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx:21]: You need it to be generic to support the follow up ticket

Comment (matheusmendes-cb) [reply src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx:21]: Yes, I realized it after seeing the other ticket. Fixing it right now.

File: src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx [hunk 3]
```diff
@@ -15,11 +16,22 @@ export default function UserActionOptInSWCDeepLink() {
...
   const session = useSession()
+  const searchParams = useSearchParams()
+
+  const { destination } = getCallbackDestination({
+    queryString: searchParams?.toString(),
+    defaultDestination: 'profile',
+  })
+
+  const handleRedirectOnLogin = React.useCallback(() => {
+    return router.replace(urls[destination]())
```
Comment (travisbloom-cb) [inline src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx:27]: can we gracefully fail/trigger sentry here?

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx (hunk 3 above)
Comment (matheusmendes-cb) [reply src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx:42]: Done

File: src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx [hunk 4]
```diff
@@ -18,14 +19,28 @@ export default function UserActionOptInSWCDeepLink() {
...
   const { destination } = getCallbackDestination({
-    queryString: searchParams?.toString(),
+    queryString,
     defaultDestination: 'profile',
   })
 
   const handleRedirectOnLogin = React.useCallback(() => {
+    // should always have a destination, but if something happens and thats no the case,
+    // we should gracefully fail, redirecting the user to the profile page
+    if (!destination) {
```
Comment (travisbloom-cb) [inline src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx:32]: the bug im looking to fix is what if there's a destination and its an invalid urls param. We actually dont wanna trigger sentry errors if the user doesnt pass destination

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx (hunk 4 above)
Comment (matheusmendes-cb) [reply src/app/[locale]/(homepageDialogDeeplink)/action/sign-up/page.tsx]: Fixed!

**Review by travisbloom-cb — APPROVED** [review]

**Pull request merged**